workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md

```diff
@@ -0,0 +1,527 @@
+* [Setup JupyterHub](https://zero-to-jupyterhub.readthedocs.io/en/latest/#setup-jupyterhub)
+* [Setup BinderHub](https://binderhub.readthedocs.io/en/latest/setup-registry.html#set-up-the-container-registry)
+
+## Cloud Resource Requirements <a name="cloudresoures"></a>
+
+This workshop assumes you have a "Free Trial" subscription with [Microsoft Azure](https://azure.microsoft.com/en-gb/).
+It's quick to set one up and you get £150 free credit for the first 30 days as well as access to some _always free_ services.
+
+> BinderHub is Cloud-neutral.
+> We are using Azure as an example.
+
+## Container Registry <a name="containerreg"></a>
+
+These instructions will link the BinderHub to a [DockerHub](https://hub.docker.com/) Container Registry, and so you will need a DockerHub account as well.
+
+> BinderHub also works with Google Container Registry and custom registries.
+> We are using DockerHub as an example.
+
+## Installation Requirements <a name="installation"></a>
+
+This workshop will use a terminal (as opposed to Azure's [Cloud Shell](https://azure.microsoft.com/en-gb/features/cloud-shell/) interface) and so we require some command line interfaces.
+
+* **Azure CLI:** Installation guidelines found [here](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
+* **Kubernetes CLI (`kubectl`):** Installation guidelines found [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-with-homebrew-on-macos)
+* **Helm (Kubernetes package manager):** Installation guidelines found [here](https://helm.sh/docs/using_helm/#installing-helm)
+
+We used Homebrew on MacOS to install these:
+```bash
+brew install azure-cli
+brew install kubernetes-cli
+brew install kubernetes-helm
+```
+
+## Deploying a Kubernetes cluster on Azure <a name="k8s"></a>
+
+Adapted from [Step Zero: Kubernetes on Microsoft Azure Container Service (AKS)](https://zero-to-jupyterhub.readthedocs.io/en/latest/microsoft/step-zero-azure.html).
+
+A short (and by no means exhaustive) [glossary](#glossary) of Kubernetes terms is given at the end of this workshop, should you require further explanation.
+
+### 1. Login to Azure <a name="aks-step1"></a>
+
+```bash
+az login
+```
+
+This command will open a browser window for you to log in to your Azure account.
+You can safely close this window after logging in.
+
+### 2. Activate your Subscription <a name="aks-step2"></a>
+
+To see a list of Azure subscriptions you have available to you, you can run the following command.
+```bash
+az account list --refresh --output table
+```
+This prints your subscriptions to the terminal in a human-readable format.
+
+You should only see a "Free Trial" subscription (unless you have used Azure before and have others).
+Let's activate this with the following command.
+```bash
+az account set -s "Free Trial"
+```
+**N.B.:** If you wish to use a different subscription, replace the text in quotes with the name of your chosen subscription.
```

Comment:
Do we need an action plan for people who have already used up their free trial?

Reply:
Possibly. What do you recommend?